receiver/mysqlreceiver/metadata.yaml

```
@@ -279,3 +287,39 @@ metrics:
       monotonic: false
       aggregation: cumulative
     attributes: [threads]
+  mysql.table_lock_wait.read.count:
+    enabled: true
+    description: The total table lock wait read events.
+    unit: "1"
+    sum:
+      value_type: int
+      monotonic: false
+      aggregation: cumulative
+    attributes: [schema, table_name, read_lock_type]
+  mysql.table_lock_wait.read.time:
+    enabled: true
+    description: The total table lock wait read events times.
+    unit: ns
+    sum:
+      value_type: int
+      monotonic: false
+      aggregation: cumulative
+    attributes: [schema, table_name, read_lock_type]
+  mysql.table_lock_wait.write.count:
+    enabled: true
+    description: The total table lock wait write events.
+    unit: "1"
+    sum:
+      value_type: int
+      monotonic: false
+      aggregation: cumulative
+    attributes: [schema, table_name, write_lock_type]
+  mysql.table_lock_wait.write.time:
+    enabled: true
+    description: The total table lock wait write events times.
+    unit: ns
+    sum:
+      value_type: int
+      monotonic: false
+      aggregation: cumulative
+    attributes: [schema, table_name, write_lock_type]
```

`mysql.table` seems appropriate for a namespace.

```suggestion
  mysql.table.lock_wait.read.count:
    enabled: true
    description: The total table lock wait read events.
    unit: "1"
    sum:
      value_type: int
      monotonic: false
      aggregation: cumulative
    attributes: [schema, table_name, read_lock_type]
  mysql.table.lock_wait.read.time:
    enabled: true
    description: The total table lock wait read events times.
    unit: ns
    sum:
      value_type: int
      monotonic: false
      aggregation: cumulative
    attributes: [schema, table_name, read_lock_type]
  mysql.table.lock_wait.write.count:
    enabled: true
    description: The total table lock wait write events.
    unit: "1"
    sum:
      value_type: int
      monotonic: false
      aggregation: cumulative
    attributes: [schema, table_name, write_lock_type]
  mysql.table.lock_wait.write.time:
    enabled: true
    description: The total table lock wait write events times.
    unit: ns
    sum:
      value_type: int
      monotonic: false
      aggregation: cumulative
    attributes: [schema, table_name, write_lock_type]
```